### PR TITLE
PT-3851 token paused event

### PR DESCRIPTION
**Pull request by alexbukata**

### CODE REVIEW

File: nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionPausedLogDescriptor.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.reactivestreams.Publisher
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import scalether.domain.Address
+import scalether.domain.response.Log
+import scalether.domain.response.Transaction
+
+@Component
+class CollectionPausedLogDescriptor(ignoredTokenResolver: IgnoredTokenResolver) : LogEventDescriptor<CollectionPaused> {
+    private val skipContracts = ignoredTokenResolver.resolve()
```
Comment (yakischik) [inline nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionPausedLogDescriptor.kt:18]: Is it SET?

File: nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionUnpausedLogDescriptor.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import reactor.core.publisher.Mono
+import scalether.domain.Address
+import scalether.domain.response.Log
+import scalether.domain.response.Transaction
+
+@Component
+class CollectionUnpausedLogDescriptor(
+    ignoredTokenResolver: IgnoredTokenResolver
+) : LogEventDescriptor<CollectionPaused> {
+    private val skipContracts = ignoredTokenResolver.resolve()
```
Comment (yakischik) [inline nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionUnpausedLogDescriptor.kt:20]: Is it SET?

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt [hunk 3]
```diff
@@ -335,6 +335,14 @@ class MongoOrderRepository(
         ).collectList().awaitFirst()
     }
 
+    override fun findNotCancelledOrdersByToken(token: Address): Flow<Order> {
```
Comment (yakischik) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:338]: Not sure this query is effective, is there appropriate index for it?

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt [hunk 4]
```diff
@@ -335,6 +335,14 @@ class MongoOrderRepository(
...
+        val criteria = Criteria().orOperator(
+            (Order::make / Asset::type / NftAssetType::token).isEqualTo(token),
+            (Order::take / Asset::type / NftAssetType::token).isEqualTo(token),
+        ).and(Order::cancelled).isEqualTo(false)
```
Comment (yakischik) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:342]: Order might be INACTIVE/NOT_STARTED, should we also check them?

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    suspend fun onCollectionChanged(
+        collection: NftCollectionDto,
+        eventTimeMarks: EventTimeMarks
+    ) {
+        val paused = collection.flags
+            ?.firstOrNull { it.flag == NftCollectionFlagDto.Flag.PAUSED }
+            ?.value?.toBooleanStrict() ?: false
+        if (paused) {
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt:45]: We decided to write in "fast fail" manner, like if (!paused) return

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                                hash.prefixed(),
+                                collection.id
+                            )
+                            orderCancelService.cancelOrder(hash, eventTimeMarks)
+                        }
+                    }.awaitAll()
+                }
+            }
+            logger.info(
+                "Cancelled {} active Orders for Collection {} which was paused ({}ms)",
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt:71]: "were paused"

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt (hunk 5 above)
Comment (alexbukata) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt:45]: ok

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt (hunk 3 above)
Comment (alexbukata) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:338]: Yes, we have two indexes
each starts with "make.type.token" and "take.type.token"
In this query mongo will use both of them. Double checked it with mongodb explain

File: nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionPausedLogDescriptor.kt (hunk 1 above)
Comment (alexbukata) [reply nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionPausedLogDescriptor.kt:18]: yes

File: nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionUnpausedLogDescriptor.kt (hunk 2 above)
Comment (alexbukata) [reply nft/listener/src/main/kotlin/com/rarible/protocol/nft/listener/service/descriptors/CollectionUnpausedLogDescriptor.kt:20]: yes

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt (hunk 4 above)
Comment (alexbukata) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:342]: done

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt (hunk 6 above)
Comment (alexbukata) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt]: I doubt, collection here is single
Does it matter for this log entry?

**Review by yakischik — APPROVED** [review]

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt (hunk 6 above)
Comment (yakischik) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/service/order/OrderCollectionService.kt]: Oops, i thought it is related to Orders

**Review by yakischik — APPROVED** [review]

**Pull request merged**